### PR TITLE
feature: Ability to customize icons; avoid large icon dependency

### PR DESCRIPTION
**Pull request by baleeds**
This PR will:
- Bump React peer dep to ensure support for Context API
- Add default icons map
- Add default Icon component
- Add config context to provide both items
- Swap all uses of hardcoded Icon component
- Add documentation for swapping icons
- Bump package version

Points of contention:
- Naming of new context, and the fact that it only contains the icon specific things, even though some other stuff feels like it belongs in the "config"
- Tests were failing when I tried them before and after my work
- Generating the file type to icon map in `IconUtils` runs every now and then, but its not the least efficient thing ever

 https://github.com/TimboKZ/Chonky/issues/5

### CODE REVIEW

**Comment by baleeds** [comment]
@audiolion @mmgolden
What do you guys think?

**Comment by TimboKZ** [comment]
The pull request looks good overall, I like the new icon abstraction. Will need to pull it and test locally.

The code formatting is now a bit inconsistent throughout the project, but I was planning to add Prettier to the project so that shouldn't matter much.

Will try to merge in the next couple of days.

**Comment by TimboKZ** [comment]
RE: Tests - ignore them, they don't do anything at the moment. I'll figure something out later.

**Comment by baleeds** [comment]
Sounds good.  Yeah I noticed the formatting, definitely muddles the review process.  I think prettier would be a very nice addition.

**Comment by audiolion** [comment]
> Sounds good. Yeah I noticed the formatting, definitely muddles the review process, sorry about that. I think prettier would be a very nice addition.

> The code formatting is now a bit inconsistent throughout the project, but I was planning to add Prettier to the project anyway so that shouldn't matter much.

Perhaps a separate PR is created and merged first that applies prettier and adds a prettier config so that when this PR is rebased it will remove all formatting changes?

File: package.json [hunk 1]
```diff
@@ -27,12 +27,13 @@
...
     "javascript-time-ago": "^2.0.1",
+    "memoizee": "^0.4.14",
     "react-if": "^3.4.3",
     "react-virtualized": "^9.21.1",
     "shallow-equal": "^1.2.0"
   },
   "peerDependencies": {
-    "react": "^16.0.0",
+    "react": "^16.3.0",
     "react-dom": "^16.0.0"
```
[the remark this answers is not in the document]
Comment (baleeds) [reply package.json]: Understood!

**Comment by TimboKZ** [comment]
Just merged #7, can you please rebase your changes on top of that?

**Comment by baleeds** [comment]
It worked!  Much easier to see what I changed now.  It was a gnarly merge resolution though, so there is possibly some broken things.  I tested everything I could think of manually in the styleguide.

File: src/components/ConfigContext.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { Icon, icons } from './Icon';
+
+export interface ConfigValue {
+  icons: typeof icons;
+  Icon: typeof Icon;
+}
+
+export const ConfigContext = React.createContext<ConfigValue>({ Icon, icons });
```
Comment (TimboKZ) [inline src/components/ConfigContext.tsx]: We should probably break this context down into smaller parts once we add hooks.

**Pull request merged**